modules/entrezdirect/esummary/main.nf

```diff
@@ -0,0 +1,52 @@
+process ENTREZDIRECT_ESUMMARY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::entrez-direct=16.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/entrez-direct:16.2--he881be0_1':
+        'quay.io/biocontainers/entrez-direct:16.2--he881be0_1' }"
+
+    input:
+    tuple val(meta), val(uid), path(uids_file)
+    val database
+
+    output:
+    tuple val(meta), path("*.xml"), emit: xml_esummary
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def input_id = uid ? "-id ${uid}" : ''
+    def input_file = uids_file.name != 'NO_FILE' ? "-input $uids_file" : ''
+    if(input_id && !input_file)
+    """
+    esummary \\
+        $args \\
+        -db $database \\
+        $input_id | cat | grep '<' > ${prefix}.xml
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        esummary: \$(echo \$(esummary --help | head -1 | cut -d' ' -f2 2>&1) | sed 's/^esummary //; s/Using.*\$//' ))
+    END_VERSIONS
+    """
+    else if(input_file && !input_id)
+    """
+    esummary \\
+        $args \\
+        -db $database \\
+        $input_file | cat | grep '<' > ${prefix}.xml
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        esummary: \$(echo \$(esummary --help | head -1 | cut -d' ' -f2 2>&1) | sed 's/^esummary //; s/Using.*\$//' ))
+    END_VERSIONS
+    """
```

Comment:
Done. I removed the command duplication. The `cat` was not required, I already removed it. When using the latest version of entrez-direct (16.2_1) there is a bug that produces an output like the following (see discussion in [issue #1771](https://github.com/nf-core/modules/issues/1771#)).


```
Can't locate Time/HiRes.pm in @INC (you may need to install the Time::HiRes module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.28.1 /usr/local/share/perl/5.28.1 /usr/lib/x86_64-linux-gnu/perl5/5.28 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.28 /usr/share/perl/5.28 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base).
BEGIN failed--compilation aborted.
<ENTREZ_DIRECT>
  <Db>pubmed</Db>
  <WebEnv>MCID_62ac9c50d7198b20634e5a32</WebEnv>
  <QueryKey>1</QueryKey>
  <Count>17387</Count>
  <Step>1</Step>
</ENTREZ_DIRECT>
```
 But the output should only be in XML format (lines starting with `<`),  to avoid these other lines, I used `grep '<'` to have the following output: 


```
<ENTREZ_DIRECT>
  <Db>pubmed</Db>
  <WebEnv>MCID_62ac9c50d7198b20634e5a32</WebEnv>
  <QueryKey>1</QueryKey>
  <Count>17387</Count>
  <Step>1</Step>
</ENTREZ_DIRECT>
```

Reply:
Ah understood - I would add a comment to that effect in the script block for future devs (I'll make a suggestion)